src/chainstate/stacks/index/file.rs

```diff
@@ -181,15 +229,194 @@ impl TrieFile {
         Ok(trie_blob)
     }
 
+    pub fn get_trie_offset(
+        &mut self,
+        db: &Connection,
+        block_id: u32,
+    ) -> Result<TrieIdOffset, Error> {
+        if let Some(cached) = self.trie_offsets.get(&block_id) {
+            Ok(TrieIdOffset {
+                offset: cached.offset,
+                length: cached.length,
+            })
+        } else {
+            let extern_trie = trie_sql::get_external_trie_offset_length(db, block_id)?;
+            let offset = TrieIdOffset {
+                offset: extern_trie.offset,
+                length: extern_trie.length,
+            };
+            self.trie_offsets.insert(block_id, offset);
+            Ok(offset)
+        }
+    }
+
+    pub fn load_trie_blob(&mut self, db: &Connection, block_id: u32) -> Result<(), Error> {
+        // If the specified block_id is the currently loaded block, simply return.
+        if let Some(current_block_id) = self.current_block_id {
+            if current_block_id == block_id {
+                return Ok(());
+            }
+        }
+
+        // Check the LRU cache for the specified block.  If found, set the loaded trie
+        // to the cached version instead of reading from disk.
+        if let Some(cached_trie) = self.decompressed_lru.get(&block_id) {
+            self.current_block_id = Some(block_id);
+            self.current_trie = Some(Cursor::new(cached_trie.to_vec()));
+            return Ok(());
+        }
+
+        // We must retrieve the trie from disk.  Retrieve the trie offset+length from the index DB,
+        // read the full contents of the trie, decompress it, cache it in the LRU, and set
+        // the currently loaded trie.
+
+        let bench_start = SystemTime::now();
+        let extern_trie = trie_sql::get_external_trie_offset_length(db, block_id)?;
+        //eprintln!("Loading trie blob with block_id={}, offset={}, length={}, compression={}",
+        //    &block_id, &extern_trie.offset, &extern_trie.length, &extern_trie.compression);
+
+        self.seek(SeekFrom::Start(extern_trie.offset))?;
+        let mut buf = vec![0u8; extern_trie.length as usize];
+        self.read_exact(&mut buf)?;
+        let result: Vec<u8>;
+
+        match extern_trie.compression {
+            BlobCompressionType::None => {
+                result = buf;
+            }
+            BlobCompressionType::LZ4 => {
+                result = lz4_decompress_size_prepended(buf.as_slice()).unwrap();
```

Comment:
We generally try to avoid using `.unwrap()` for fallible code.  Can you elaborate under what circumstances this function can fail?  Even if it fails via an irrecoverable error, the user of `.expect()` is preferred because then you at least get an error message.